spring-kafka/src/main/java/org/springframework/kafka/listener/CommonErrorHandler.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+import java.util.List;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+
+import org.springframework.kafka.support.TopicPartitionOffset;
+
+/**
+ * Replacement for {@link ErrorHandler} and {@link BatchErrorHandler} and their
+ * sub-interfaces.
+ *
+ * @author Gary Russell
+ * @since 2.7.4
```

Comment:
I'm confused a little bit: the issue says `3.0 Backlog`, but looks like you even would like to back-port it to `2.7.x`.
Although I may see your point: you want to introduce it now to deprecate and remove those gradually in the future versions.
What is you thought  on the matter, please?

Reply:
Oops - sorry; I started this work a few weeks ago before I decided to make main 2.8.x.

I also want to make a few more changes to ease migration of the old error handlers.